app/lib/groups/group_permission.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'package:group_domain_models/group_domain_models.dart';
+import 'package:meta/meta.dart';
+
+enum GroupPermission {
+  /// Used for administrative tasks like e.g. editing course names/settings or
+  /// editing homeworks created by another user in the group.
+  administration,
+
+  /// Used to see if a user can create content (homeworks, info sheets, etc).
+  /// In some places in the code this is also used to check if certain content
+  /// can be edited, even if created by other users (timetable entry).
+  /// There might be more usages than just strictly content creation.
```

Comment:
I think it would be good to follow here the Effective Dart guidlines: https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

> DO start doc comments with a single-sentence summary.
>
> Start your doc comment with a brief, user-centric description ending with a period. A sentence fragment is often sufficient. Provide just enough context for the reader to orient themselves and decide if they should keep reading or look elsewhere for the solution to their problem.